src/main/scala/io/opentargets/etl/common/ConsumeWhileIterator.scala

```diff
@@ -0,0 +1,25 @@
+package io.opentargets.etl.common
+
+import scala.annotation.tailrec
+
+class ConsumeWhileIterator[A, B](iter: Iterator[A]) {
```

Comment:
This is more or less a library utility and should have some documentation on how it is used and why.

Reply:
Added some draft wording myself!